ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -100,7 +103,22 @@ object ParallelIndexerFactory {
                   size = maxInputBufferSize,
                 )
                 .map(_ -> System.nanoTime())
-            ).map(_ => ())
+            )
+              .map(_ => ())
+              .keepAlive(
+                keepAliveMaxIdleDuration,
+                () =>
+                  if (dbDispatcher.currentHealth() == HealthStatus.healthy) {
+                    logger.debug("Indexer keep-alive: database connectivity OK")
+                    ()
```

Comment:
Minor: doesn't `logger.debug` already return `Unit`? If this is caused by a warning feel free to ignore this comment.
```suggestion
                    logger.debug("Indexer keep-alive: database connectivity OK")
```